api_app/serializers.py

```diff
@@ -43,6 +43,31 @@ def get_permissions_map(self, created):
         }
 
 
+class JobAvailabilitySerializer(serializers.ModelSerializer):
+    """
+    Serializer for ask_analysis_availability
+    """
+
+    class Meta:
+        model = Job
+        fields = "__all__"
+
+    md5 = serializers.CharField(max_length=128, required=True)
+    analyzers_needed = serializers.ListField(default=list)
+    run_all_available_analyzers = serializers.BooleanField(default=False)
+    running_only = serializers.BooleanField(default=False)
+
+    def validate(self, data) -> dict:
```

Comment:
also validate/check for the case that if `run_all = True` then `analyzers_needed` should be empty.